readthedocs/doc_builder/python_environments.py

```diff
@@ -253,12 +254,13 @@ def install_user_requirements(self):
             builder_class = get_builder_class(self.project.documentation_type)
             docs_dir = (builder_class(build_env=self.build_env, python_env=self)
                         .docs_dir())
-            for path in [docs_dir, '']:
-                for req_file in ['pip_requirements.txt', 'requirements.txt']:
-                    test_path = os.path.join(self.checkout_path, path, req_file)
-                    if os.path.exists(test_path):
-                        requirements_file_path = test_path
-                        break
+            paths = [docs_dir, '']
+            req_files = ['pip_requirements.txt', 'requirements.txt']
+            for path, req_file in itertools.product(paths, req_files):
```

Comment:
I don't see any logic change here, right?

We just had 2 nested for and now we use `itertools.products` that produces the same result in the end. Am I right?

Reply:
```
>>> import itertools
>>> d = ['a', 'b']
>>> r = [1, 2]
>>> iter
iter(      itertools  
>>> list(itertools.product(d, r))
[('a', 1), ('a', 2), ('b', 1), ('b', 2)]
>>> for _d in d:
...     for _r in r:
...         print(_d, _r)
... 
a 1
a 2
b 1
b 2
>>> 
````

Reply:
It is the same result, but the break now exists the loop https://github.com/rtfd/readthedocs.org/issues/2812#issuecomment-372476866

Reply:
Oh boy! Great catch! :)